lib/phoenix_live_view/html_engine.ex

```diff
@@ -762,4 +765,30 @@ defmodule Phoenix.LiveView.HTMLEngine do
   defp void?(_), do: false
 
   defp to_location(%{line: line, column: column}), do: [line: line, column: column]
+
+  @attrs_with_required_ids ["phx-update", "phx-hook"]
+
+  # Check if attrs contains `@attrs_with_required_ids`.
+  #
+  # In case one is found and there is no ID, an error will be raised. Otherwise,
+  # it returns :ok.
+  defp validate_phx_attrs!(attrs, meta, state),
+    do: validate_phx_attrs!(attrs, meta, state, %{id: nil, attr: nil})
```

Comment:
You don't use the `id` attribute, so I would change it to this:

```suggestion
    do: validate_phx_attrs!(attrs, meta, state, nil)
```